### PR TITLE
Drop labels in Image and Audio folders if files are on different levels in directory or if there is only one label

### PR DESCRIPTION
**Pull request by polinaeterna**
Will close https://github.com/huggingface/datasets/issues/5153

Drop labels by default (`drop_labels=None`) when:
* there are files on different levels of directory hierarchy by checking their path depth
* all files are in the same directory (=only one label was inferred)

First one fixes cases like this:
```
repo
   image3.jpg
   image4.jpg
   data
      image1.jpg
      image2.jpg
```
Second one fixes cases like this:
```
repo
   image1.jpg
   image2.jpg
   image3.jpg
```
This is mostly to fix the viewer for people who just drop images in the Hub interface into the root dir.

I added tests for both of the cases on local and remote files. **I also changed data files for old test on drop_labels** (`test_generate_examples_drop_labels`). The files I provide to `test_generate_examples_drop_labels` now has "canonical" classification structure (two dirs) in order not to change the logic of the test (=not to check these two cases addressed in the PR).

### CODE REVIEW

**Review by lhoestq — COMMENTED** [review]
Awesome ! It looks all good to me :)

Feel free to mark your PR as "ready for review"

**Review by mariosasko — APPROVED** [review]
Thanks, LGTM!

Nit:  maybe we can use the `count_path_segments` function from this file for counting (updated with your logic to make it faster).

**Review by lhoestq — APPROVED** [review]
It looks all good, thanks !

**Comment by polinaeterna** [comment]
> Nit: maybe we can use the count_path_segments function from this file for counting (updated with your logic to make it faster).

@mariosasko just to make sure I understood you correctly - are you okay with this change? (actually `os.path.normpath` is redundant here as paths from `data_files` should be already normalized but just in case)
https://github.com/huggingface/datasets/pull/5192/files#diff-1f09f7a178211f7539b1499b64b69793bd53b30c8b7b34cfcc5835e25d31929fR33
If you are, we can merge.

File: src/datasets/packaged_modules/folder_based_builder/folder_based_builder.py [hunk 1]
```diff
@@ -30,17 +31,9 @@ def pa_table_to_pylist(table):
...
-            break
-        elif parts[1] == path:
-            break
-        else:
-            path = parts[0]
-            cnt += 1
-    return cnt
+    if is_remote_url(path):
+        return path.count("/")
+    return os.path.normpath(path).count(os.sep)
```
Comment (mariosasko) [inline src/datasets/packaged_modules/folder_based_builder/folder_based_builder.py:36]: Yes, we resolve the paths in `data_files.py`, so `normpath` is redundant here.

This should also work, no:
```suggestion
    return path.replace("\\", "/").count("/")
```

**Review by mariosasko — COMMENTED** [review]
One nit. Besides that, looks all good.

**Comment by lhoestq** [comment]
awesome ! :D

**Pull request merged**